csv/src/main/java/org/apache/metamodel/csv/CsvWriter.java

```diff
@@ -72,23 +72,38 @@ public String buildLine(String[] nextLine) {
     }
 
     private boolean stringContainsSpecialCharacters(String line) {
-        return line.indexOf(_configuration.getQuoteChar()) != -1 || line.indexOf(_configuration.getEscapeChar()) != -1;
+        boolean result = line.indexOf(_configuration.getQuoteChar()) != -1
+                || line.indexOf(_configuration.getEscapeChar()) != -1;
```

Comment:
Actually .indexOf(...) is better here because it is a char and the search time of ```indexOf(char)``` is presumably a lot faster than ```contains(String)``` since it only has to evaluate a single character.